### DISCUSSION TITLE
How to incremental download Inbox email with ImapClient Class.

### DISCUSSION THREAD

**Discussion by shenglingshi** [jstedfast/MimeKit#992] [General]
I have a EmailClient, use ImapClient to download Inbox email.
But I don't want use this code to download email.

```
public static void DownloadMessages ()
{
    using (var client = new ImapClient ()) {
        client.Connect ("imap.gmail.com", 993, SecureSocketOptions.SslOnConnect);

        client.Authenticate ("username", "password");

        client.Inbox.Open (FolderAccess.ReadOnly);

        var uids = client.Inbox.Search (SearchQuery.All);

        foreach (var uid in uids) {
            var message = client.Inbox.GetMessage (uid);

            // write the message to a file
            message.WriteTo (string.Format ("{0}.eml", uid));
        }

        client.Disconnect (true);
    }
}
```

Because this client.Inbox.Search (SearchQuery.All)  get all inbox email every time .
I want download email incremental ,But I can't find a good SearchQuery. 
Anybody have a good advice.

**Comment by jstedfast** [comment]
The IMAP server doesn't track which messages your client has already downloaded, so there is no Search query that will give you what you want.

You will need to track which messages your client has downloaded and start where it left off.

**Reply by shenglingshi** [reply]
Ok ,Get It.Tks.
